--- /clang/lib/Driver/ToolChains/CommonArgs.cpp
+++ /clang/lib/Driver/ToolChains/CommonArgs.cpp
@@ -1256,7 +1256,8 @@
 
   LibGccType LGT = getLibGccType(D, Args);
   bool AsNeeded = LGT == LibGccType::UnspecifiedLibGcc &&
-                  !TC.getTriple().isAndroid() && !TC.getTriple().isOSCygMing();
+                  !TC.getTriple().isAndroid() && !TC.getTriple().isOSCygMing() ||
+                  UNW == ToolChain::UNW_CompilerRT;
   if (AsNeeded)
     CmdArgs.push_back("--as-needed");
 
