--- a/arch/x86/include/asm/cpufeatures.h	2021-10-31 21:53:10.000000000 +0100
+++ b/arch/x86/include/asm/cpufeatures.h	2021-12-11 21:18:48.369230918 +0100
@@ -312,6 +312,7 @@
 #define X86_FEATURE_AMD_PPIN		(13*32+23) /* Protected Processor Inventory Number */
 #define X86_FEATURE_AMD_SSBD		(13*32+24) /* "" Speculative Store Bypass Disable */
 #define X86_FEATURE_VIRT_SSBD		(13*32+25) /* Virtualized Speculative Store Bypass Disable */
+#define X86_FEATURE_CPPC               (13*32+27) /* Collaborative Processor Performance Control */
 #define X86_FEATURE_AMD_SSB_NO		(13*32+26) /* "" Speculative Store Bypass is fixed in hardware. */
 
 /* Thermal and Power Management Leaf, CPUID level 0x00000006 (EAX), word 14 */
