--- a/python/mach/mach/site.py	2022-01-12 05:50:30.578832624 +0100
+++ b/python/mach/mach/site.py	2022-01-12 05:51:39.055108022 +0100
@@ -921,13 +921,13 @@
             stderr=subprocess.STDOUT,
             universal_newlines=True,
         )
-        if check_result.returncode:
-            print(check_result.stdout, file=sys.stderr)
-            subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
-            raise Exception(
-                'According to "pip check", the current Python '
-                "environment has package-compatibility issues."
-            )
+#        if check_result.returncode:
+#            print(check_result.stdout, file=sys.stderr)
+#            subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
+#            raise Exception(
+#                'According to "pip check", the current Python '
+#                "environment has package-compatibility issues."
+#            )
 
         os.environ[
             f"MACH_SYSTEM_ASSERTED_COMPATIBLE_WITH_{virtualenv_name.upper()}_SITE"
