--- mm/internal.h	2021-12-08 19:25:44.616149960 +0100
+++ mm/internal.h-mod	2021-12-08 20:27:36.401482346 +0100
@@ -36,9 +36,9 @@
 
 vm_fault_t do_swap_page(struct vm_fault *vmf);
 
-void free_pgtables(struct mmu_gather *tlb, struct vm_area_struct *start_vma,
-		unsigned long floor, unsigned long ceiling);
-
+void free_pgtables(struct mmu_gather *tlb, struct maple_tree *mt,
+                  struct vm_area_struct *start_vma, unsigned long floor,
+                  unsigned long ceiling);
 static inline bool can_madv_lru_vma(struct vm_area_struct *vma)
 {
 	return !(vma->vm_flags & (VM_LOCKED|VM_HUGETLB|VM_PFNMAP));
