--- a/net/core/sock_map.c	2021-10-31 21:53:10.000000000 +0100
+++ b/net/core/sock_map.c	2021-11-26 13:48:57.312783279 +0100
@@ -511,11 +511,11 @@
 	       ops->op == BPF_SOCK_OPS_TCP_LISTEN_CB;
 }
 
-static bool sk_is_tcp(const struct sock *sk)
-{
-	return sk->sk_type == SOCK_STREAM &&
-	       sk->sk_protocol == IPPROTO_TCP;
-}
+//static bool sk_is_tcp(const struct sock *sk)
+//{
+//	return sk->sk_type == SOCK_STREAM &&
+//	       sk->sk_protocol == IPPROTO_TCP;
+//}
 
 static bool sock_map_redirect_allowed(const struct sock *sk)
 {
