--- a/lib/string.c	2021-11-26 13:53:23.416612201 +0100
+++ b/lib/string.c	2021-11-26 13:58:14.384677449 +0100
@@ -51,23 +51,6 @@
 	uintptr_t as_uptr;
 };
 
-#define BYTES_LONG	sizeof(long)
-#define WORD_MASK	(BYTES_LONG - 1)
-#define MIN_THRESHOLD	(BYTES_LONG * 2)
-
-/* convenience union to avoid cast between different pointer types */
-union types {
-	u8 *as_u8;
-	unsigned long *as_ulong;
-	uintptr_t as_uptr;
-};
-
-union const_types {
-	const u8 *as_u8;
-	const unsigned long *as_ulong;
-	uintptr_t as_uptr;
-};
-
 #ifndef __HAVE_ARCH_STRNCASECMP
 /**
  * strncasecmp - Case insensitive, length-limited string comparison
